### PR TITLE
Tanks are able to replace conduits

### PR DESCRIPTION
**Pull request by Slava0135**
Peek 2020-12-11 10-24

I it is always irritating me that you have to remove conduits before placing a tank - the liquid is going to leak. Also when the said liquid is slag it might catch on fire the conduit.

### CODE REVIEW

File: core/src/mindustry/world/blocks/liquid/LiquidRouter.java [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 import mindustry.gen.*;
 import mindustry.type.*;
+import mindustry.world.Block;
+import mindustry.world.meta.BlockGroup;
```
Comment (summetdev) [inline core/src/mindustry/world/blocks/liquid/LiquidRouter.java:6]: use wildcard imports:

```suggestion
import mindustry.world.*;
import mindustry.world.meta.*;
```

**Review by summetdev — CHANGES_REQUESTED** [review]

File: core/src/mindustry/world/blocks/liquid/LiquidRouter.java [hunk 2]
```diff
@@ -24,4 +26,10 @@ public boolean acceptLiquid(Building source, Liquid liquid){
             return (liquids.current() == liquid || liquids.currentAmount() < 0.2f);
         }
     }
+
+    @Override
+    public boolean canReplace(Block other){
+        if(other.alwaysReplace) return true;
+        return this.group != BlockGroup.none && other.group == this.group && other instanceof Conduit;
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/liquid/LiquidRouter.java:33]: This prevents tanks from replacing liquid routers.

**Comment by Slava0135** [comment]
so now it works as it should

**Comment by Slava0135** [comment]
Peek 2020-12-11 19-17

**Comment by Anuken** [comment]
Using `instanceof` checks still seems messy, I'll implement this with special group flags.

**Comment by Slava0135** [comment]
ok then

**Pull request closed without merging**